### ISSUE TITLE
Googled drive backup not working

### ISSUE THREAD

**Issue by michaelthompson1991** [sabeechen/hassio-google-drive-backup#756]
###### Description:
```
If you have anything else that could help explain what happened, click "Markdown" above and write it here.
```

 Addon version: `0.109.2`
 Home Assistant Version: `2022.11.4`
 Supervisor Version: `2022.11.2`
 Supervisor Channel: `stable`
 Hassos Version: `9.3`
 Docker Version: `20.10.18`
 Architecture: `aarch64`
 Machine: `raspberrypi3-64`
 Date: `2022-11-26T20:13:40.769909+00:00`
 Timezone: `Europe/London`
 Failure Time: `2022-11-26T20:13:12.398689+00:00`
 Last Good Sync: `2022-11-26T20:12:56.222022+00:00`
 ###### Exception:
 ```
 
  addon/backup/model/coordinator.py:177 (_sync)
  addon/backup/model/model.py:196 (sync)
  addon/backup/model/model.py:304 (_syncBackups)
  addon/backup/drive/drivesource.py:118 (get)
  addon/backup/drive/drivesource.py:269 (getFolderId)
  addon/backup/drive/folderfinder.py:65 (get)
  addon/backup/drive/folderfinder.py:117 (_readFolderId)
  addon/backup/drive/folderfinder.py:158 (_verify)
  addon/backup/drive/driverequests.py:165 (get)
  addon/backup/drive/driverequests.py:356 (retryRequest)
  addon/backup/drive/driverequests.py:95 (_getHeaders)
  addon/backup/drive/driverequests.py:154 (getToken)
  addon/backup/creds/exchanger.py:85 (refresh)
  addon/backup/creds/exchanger.py:159 (_refresh_default)
backup.exceptions.exceptions.CredRefreshMyError: Couldn't communicate with habackup.io

 ```
 Backups:
 ```
 {
    "HomeAssistant": {
        "backups": 8,
        "retained": 2,
        "deletable": 6,
        "name": "HomeAssistant",
        "title": "Home Assistant",
        "latest": "2022-11-26T19:44:57Z",
        "max": 5,
        "enabled": true,
        "icon": "home-assistant",
        "ignored": 0,
        "detail": "",
        "size": "63.1 MB",
        "ignored_size": "0.0 B",
        "free_space": "18.5 GB"
    },
    "GoogleDrive": {
        "backups": 6,
        "retained": 1,
        "deletable": 5,
        "name": "GoogleDrive",
        "title": "Google Drive",
        "latest": "2022-11-26T00:00:01Z",
        "max": 5,
        "enabled": true,
        "icon": "google-drive",
        "ignored": 0,
        "detail": "michaelthompson1991@me.com",
        "size": "44.0 MB",
        "ignored_size": "0.0 B",
        "free_space": "15.0 GB"
    }
}
 ```
 ###### Config:
 ```
 {
    "Setting.MAX_BACKUPS_IN_HA": 5,
    "Setting.MAX_BACKUPS_IN_GOOGLE_DRIVE": 5,
    "Setting.DAYS_BETWEEN_BACKUPS": 1.0,
    "Setting.IGNORE_UPGRADE_BACKUPS": false,
    "Setting.DELETE_BEFORE_NEW_BACKUP": true,
    "Setting.BACKUP_TIME_OF_DAY": "00:00"
}
 ```
 ###### Addon Logs:
 ```
 11-26 19:51:50 INFO [backup.model.coordinator] I'll try again in an hour
11-26 19:51:56 DEBUG [backup.ha.harequests] Making Hassio request: http://supervisor/addons/self/info
11-26 19:51:56 DEBUG [backup.ha.harequests] Making Hassio request: http://supervisor/info
11-26 19:51:56 DEBUG [backup.ha.harequests] Making Hassio request: http://supervisor/core/info
11-26 19:51:56 DEBUG [backup.ha.harequests] Making Hassio request: http://supervisor/supervisor/info
11-26 19:51:56 DEBUG [backup.ha.harequests] Making Hassio request: http://supervisor/addons
11-26 19:52:00 DEBUG [backup.ha.harequests] Making Hassio request: http://supervisor/addons/self/info
11-26 19:52:00 DEBUG [backup.ha.harequests] Making Hassio request: http://supervisor/info
11-26 19:52:00 DEBUG [backup.ha.harequests] Making Hassio request: http://supervisor/core/info
11-26 19:52:00 DEBUG [backup.ha.harequests] Making Hassio request: http://supervisor/supervisor/info
11-26 19:52:00 DEBUG [backup.ha.harequests] Making Hassio request: http://supervisor/addons
11-26 19:54:30 DEBUG [backup.ha.harequests] Making Hassio request: http://supervisor/addons/self/info
11-26 19:54:30 DEBUG [backup.ha.harequests] Making Hassio request: http://supervisor/info
11-26 19:54:30 DEBUG [backup.ha.harequests] Making Hassio request: http://supervisor/core/info
11-26 19:54:30 DEBUG [backup.ha.harequests] Making Hassio request: http://supervisor/supervisor/info
11-26 19:54:30 DEBUG [backup.ha.harequests] Making Hassio request: http://supervisor/addons
11-26 19:54:52 DEBUG [backup.ha.harequests] Making Hassio request: http://supervisor/addons/self/info
11-26 19:54:52 DEBUG [backup.ha.harequests] Making Hassio request: http://supervisor/info
11-26 19:54:52 DEBUG [backup.ha.harequests] Making Hassio request: http://supervisor/core/info
11-26 19:54:52 DEBUG [backup.ha.harequests] Making Hassio request: http://supervisor/supervisor/info
11-26 19:54:52 DEBUG [backup.ha.harequests] Making Hassio request: http://supervisor/addons
11-26 20:12:56 INFO [backup.model.coordinator] Syncing Backups
11-26 20:12:56 DEBUG [backup.ha.harequests] Making Hassio request: http://supervisor/supervisor/info
11-26 20:12:56 DEBUG [backup.ha.harequests] Making Hassio request: http://supervisor/backups
11-26 20:12:56 DEBUG [backup.drive.driverequests] Requesting refreshed Google Drive credentials
11-26 20:13:04 WARNING [backup.creds.exchanger] Unable to reach token1.habackup.io, trying alternate server(s)...
11-26 20:13:12 WARNING [backup.creds.exchanger] Unable to reach habackup.io, trying alternate server(s)...
11-26 20:13:12 ERROR [backup.creds.exchanger] Unable to refresh credentials with Google Drive
11-26 20:13:12 ERROR [backup.model.coordinator] Couldn't refresh Google Drive credentials because: Couldn't communicate with habackup.io
11-26 20:13:12 INFO [backup.model.coordinator] I'll try again in an hour
 ```
 ###### Supervisor Logs:
 ```
 [32m22-11-26 19:46:33 INFO (MainThread) [supervisor.api.middleware.security] /backups access from cebe7a76_hassio_google_drive_backup[0m
[33m22-11-26 19:48:13 WARNING (MainThread) [supervisor.api.ingress] No valid ingress session None[0m
[32m22-11-26 19:50:37 INFO (MainThread) [supervisor.api.middleware.security] /core/info access from cebe7a76_hassio_google_drive_backup[0m
[32m22-11-26 19:50:37 INFO (MainThread) [supervisor.api.middleware.security] /supervisor/info access from cebe7a76_hassio_google_drive_backup[0m
[32m22-11-26 19:50:37 INFO (MainThread) [supervisor.api.middleware.security] /addons access from cebe7a76_hassio_google_drive_backup[0m
[32m22-11-26 19:51:34 INFO (MainThread) [supervisor.api.middleware.security] /supervisor/info access from cebe7a76_hassio_google_drive_backup[0m
[32m22-11-26 19:51:34 INFO (MainThread) [supervisor.api.middleware.security] /backups access from cebe7a76_hassio_google_drive_backup[0m
[32m22-11-26 19:51:56 INFO (MainThread) [supervisor.api.middleware.security] /core/info access from cebe7a76_hassio_google_drive_backup[0m
[32m22-11-26 19:51:56 INFO (MainThread) [supervisor.api.middleware.security] /supervisor/info access from cebe7a76_hassio_google_drive_backup[0m
[32m22-11-26 19:51:56 INFO (MainThread) [supervisor.api.middleware.security] /addons access from cebe7a76_hassio_google_drive_backup[0m
[32m22-11-26 19:52:00 INFO (MainThread) [supervisor.api.middleware.security] /core/info access from cebe7a76_hassio_google_drive_backup[0m
[32m22-11-26 19:52:00 INFO (MainThread) [supervisor.api.middleware.security] /supervisor/info access from cebe7a76_hassio_google_drive_backup[0m
[32m22-11-26 19:52:00 INFO (MainThread) [supervisor.api.middleware.security] /addons access from cebe7a76_hassio_google_drive_backup[0m
[32m22-11-26 19:52:25 INFO (MainThread) [supervisor.api.middleware.security] /addons/a0d7b954_vscode/icon access from cebe7a76_hassio_google_drive_backup[0m
[32m22-11-26 19:52:25 INFO (MainThread) [supervisor.api.middleware.security] /addons/cebe7a76_hassio_google_drive_backup/icon access from cebe7a76_hassio_google_drive_backup[0m
[32m22-11-26 19:52:25 INFO (MainThread) [supervisor.api.middleware.security] /addons/core_configurator/icon access from cebe7a76_hassio_google_drive_backup[0m
[32m22-11-26 19:52:25 INFO (MainThread) [supervisor.api.middleware.security] /addons/core_ssh/icon access from cebe7a76_hassio_google_drive_backup[0m
[33m22-11-26 19:53:13 WARNING (MainThread) [supervisor.api.ingress] No valid ingress session None[0m
[32m22-11-26 19:54:30 INFO (MainThread) [supervisor.api.middleware.security] /core/info access from cebe7a76_hassio_google_drive_backup[0m
[32m22-11-26 19:54:30 INFO (MainThread) [supervisor.api.middleware.security] /supervisor/info access from cebe7a76_hassio_google_drive_backup[0m
[32m22-11-26 19:54:30 INFO (MainThread) [supervisor.api.middleware.security] /addons access from cebe7a76_hassio_google_drive_backup[0m
[32m22-11-26 19:54:50 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token[0m
[32m22-11-26 19:54:52 INFO (MainThread) [supervisor.api.middleware.security] /core/info access from cebe7a76_hassio_google_drive_backup[0m
[32m22-11-26 19:54:52 INFO (MainThread) [supervisor.api.middleware.security] /supervisor/info access from cebe7a76_hassio_google_drive_backup[0m
[32m22-11-26 19:54:52 INFO (MainThread) [supervisor.api.middleware.security] /addons access from cebe7a76_hassio_google_drive_backup[0m
[33m22-11-26 19:55:29 WARNING (MainThread) [supervisor.api.ingress] No valid ingress session None[0m
[32m22-11-26 20:12:56 INFO (MainThread) [supervisor.api.middleware.security] /supervisor/info access from cebe7a76_hassio_google_drive_backup[0m
[32m22-11-26 20:12:56 INFO (MainThread) [supervisor.api.middleware.security] /backups access from cebe7a76_hassio_google_drive_backup[0m
[32m22-11-26 20:13:40 INFO (MainThread) [supervisor.api.middleware.security] /supervisor/logs access from cebe7a76_hassio_google_drive_backup[0m

 ```
 ###### Home Assistant Core Logs:
 ```
 aiodns.error.DNSError: (11, 'Could not contact DNS servers')[0m
[31m2022-11-26 19:57:30.208 ERROR (MainThread) [metno] Access to https://aa015h6buqvih86i1.api.met.no/weatherapi/locationforecast/2.0/complete returned error 'ClientConnectorError'[0m
[33m2022-11-26 19:57:53.070 WARNING (MainThread) [homeassistant.config_entries] Config entry 'eero' for upnp integration not ready yet: Error connecting to device at location: http://192.168.4.1:1900/igd.xml, err: ('ClientConnectorError(ConnectionKey(host=\'192.168.4.1\', port=1900, is_ssl=False, ssl=None, proxy=None, proxy_auth=None, proxy_headers_hash=8035787646706138794), OSError(113, "Connect call failed (\'192.168.4.1\', 1900)"))', None); Retrying in background[0m
[31m2022-11-26 19:57:55.231 ERROR (MainThread) [metno] Access to https://aa015h6buqvih86i1.api.met.no/weatherapi/locationforecast/2.0/complete returned error 'ClientConnectorError'[0m
[31m2022-11-26 19:58:03.573 ERROR (MainThread) [custom_components.hacs] Request exception for 'https://api.github.com/rate_limit' with - Cannot connect to host api.github.com:443 ssl:default [Try again]
Traceback (most recent call last):
  File "/config/custom_components/hacs/base.py", line 457, in async_can_update
    response = await self.async_github_api_method(self.githubapi.rate_limit)
  File "/config/custom_components/hacs/base.py", line 511, in async_github_api_method
    raise HacsException(_exception)
custom_components.hacs.exceptions.HacsException: Request exception for 'https://api.github.com/rate_limit' with - Cannot connect to host api.github.com:443 ssl:default [Try again][0m
[31m2022-11-26 19:58:40.253 ERROR (MainThread) [metno] Access to https://aa015h6buqvih86i1.api.met.no/weatherapi/locationforecast/2.0/complete returned error 'ClientConnectorError'[0m
[31m2022-11-26 20:00:05.273 ERROR (MainThread) [metno] Access to https://aa015h6buqvih86i1.api.met.no/weatherapi/locationforecast/2.0/complete returned error 'ClientConnectorError'[0m
[31m2022-11-26 20:01:30.295 ERROR (MainThread) [metno] Access to https://aa015h6buqvih86i1.api.met.no/weatherapi/locationforecast/2.0/complete returned error 'ClientConnectorError'[0m
[31m2022-11-26 20:02:55.314 ERROR (MainThread) [metno] Access to https://aa015h6buqvih86i1.api.met.no/weatherapi/locationforecast/2.0/complete returned error 'ClientConnectorError'[0m
[31m2022-11-26 20:04:20.332 ERROR (MainThread) [metno] Access to https://aa015h6buqvih86i1.api.met.no/weatherapi/locationforecast/2.0/complete returned error 'ClientConnectorError'[0m
[31m2022-11-26 20:05:45.353 ERROR (MainThread) [metno] Access to https://aa015h6buqvih86i1.api.met.no/weatherapi/locationforecast/2.0/complete returned error 'ClientConnectorError'[0m
[31m2022-11-26 20:07:10.373 ERROR (MainThread) [metno] Access to https://aa015h6buqvih86i1.api.met.no/weatherapi/locationforecast/2.0/complete returned error 'ClientConnectorError'[0m
[31m2022-11-26 20:08:03.576 ERROR (MainThread) [custom_components.hacs] Request exception for 'https://api.github.com/rate_limit' with - Cannot connect to host api.github.com:443 ssl:default [Try again]
Traceback (most recent call last):
  File "/config/custom_components/hacs/base.py", line 457, in async_can_update
    response = await self.async_github_api_method(self.githubapi.rate_limit)
  File "/config/custom_components/hacs/base.py", line 511, in async_github_api_method
    raise HacsException(_exception)
custom_components.hacs.exceptions.HacsException: Request exception for 'https://api.github.com/rate_limit' with - Cannot connect to host api.github.com:443 ssl:default [Try again][0m
[31m2022-11-26 20:08:35.393 ERROR (MainThread) [metno] Access to https://aa015h6buqvih86i1.api.met.no/weatherapi/locationforecast/2.0/complete returned error 'ClientConnectorError'[0m
[31m2022-11-26 20:10:00.414 ERROR (MainThread) [metno] Access to https://aa015h6buqvih86i1.api.met.no/weatherapi/locationforecast/2.0/complete returned error 'ClientConnectorError'[0m
[31m2022-11-26 20:11:25.435 ERROR (MainThread) [metno] Access to https://aa015h6buqvih86i1.api.met.no/weatherapi/locationforecast/2.0/complete returned error 'ClientConnectorError'[0m
[31m2022-11-26 20:12:50.456 ERROR (MainThread) [metno] Access to https://aa015h6buqvih86i1.api.met.no/weatherapi/locationforecast/2.0/complete returned error 'ClientConnectorError'[0m

 ```

**Comment by sabeechen** [comment]
Based on your logs form Home Assistant, it looks like your install can't access the internet.  You're getting a bunch of errors from various components about not being able to reach websites or resolve reach your DNS server.  The addon wont be able to work without an internet connection.

If you think there is something I misunderstood, please reopen the issue and let me know how I can help.

**Comment by michaelthompson1991** [comment]
Apologies I’m nee to home assistant, here’s what I’m getting on the google drive backup Sent from my iPhoneOn 27 Nov 2022, at 00:54, Stephen Beechen ***@***.***> wrote:﻿
Closed #756 as completed.
